OSDOCS-12056: Documented the 4.13.51 z-stream RNs (#82221)
openshift/openshift-docs: release_notes/ocp-4-13-release-notes.adoc

[id="ocp-4-13-50-updating"]
==== Updating
To update an existing {product-title} 4.13 cluster to this latest release, see xref:../updating/updating-cluster-cli.adoc#updating-cluster-cli[Updating a cluster using the CLI].

// 4.13.51
[id="ocp-4-13-51"]
=== RHSA-2024:6811 - {product-title} 4.13.51 bug fix and security updates

Issued: 25 September 2024

> {product-title} release 4.13.51, which includes security updates, is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHSA-2024:6811[RHSA-2024:6811] advisory. TThe RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHBA-2024:6813[RHBA-2024:6813] advisory.

Reviewer: Typo:
```suggestion
{product-title} release 4.13.51, which includes security updates, is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHSA-2024:6811[RHSA-2024:6811] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHBA-2024:6813[RHBA-2024:6813] advisory.
```

Author: :heavy_check_mark:

Space precluded documenting all of the container images for this release in the advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.13.51 --pullspecs
----

[id="ocp-4-13-51-enhancements"]
==== Enhancements

> * CENTOS 8 has recently ended its lifecycle. This release updates the CENTOS 8 references to CENTOS 9. (link:https://issues.redhat.com/browse/OCPBUGS-39160[*OCPBUGS-39160*])

Author: * CENTOS 8 has recently ended its lifecycle. This release updates the CENTOS 8 references to CENTOS 9. (link:https://issues.redhat.com/browse/OCPBUGS-39160[*OCPBUGS-39160*]) will be removed after the peer review as the note is not needed for 4.17. 4.16 already mentions this note.

Reviewer: Wasn't sure what is meant in this comment about 4.17 since this is a 4.13.z update (maybe a copypasta leftover), but...

In general I'd be interested for more details on which "the CentOS 8 references" this is referring to. From the linked code PR in the Jira, it seems like this might make sense, if technically accurate:
```suggestion
* CentOS 8 has recently ended its lifecycle. This release updates the CentOS 8 references in the {product-title} installer to CENTOS 9. (link:https://issues.redhat.com/browse/OCPBUGS-39160[*OCPBUGS-39160*])
```
If this or a similar tweak is adopted, suggest applying to the other 4.y.z versions of this note in the related RNs as well.

Author: Sorry, confusion on my part. I had 4.17 living rent free in my mind (a term I've been meaning to use for some time). I updated the note as per your request and added the note back in. I've messaged the engineer just to confirm to re-addition of the note.

> * CENTOS 8 has recently ended its lifecycle. This release updates the CENTOS 8 references to CENTOS 9. (link:https://issues.redhat.com/browse/OCPBUGS-39160[*OCPBUGS-39160*])

Reviewer: Capitalization should be "CentOS":
```suggestion
* CentOS 8 has recently ended its lifecycle. This release updates the CentOS 8 references to CentOS 9. (link:https://issues.redhat.com/browse/OCPBUGS-39160[*OCPBUGS-39160*])
```
Out of scope nit: Would also suggest getting the capitalization fixed in the published [4.16](https://docs.openshift.com/container-platform/4.16/release_notes/ocp-4-16-release-notes.html#ocp-4-16-10-enhancements-_release-notes) and [4.14](https://docs.openshift.com/container-platform/4.14/release_notes/ocp-4-14-release-notes.html#ocp-4-16-36-enhancements-_release-notes) versions of this note next time there is a PR in progress for those. It seems like the `id`s for those 2 entries are also a little busted, FWIW (e.g., `"ocp-4-16-10-enhancements-_{context}"`).

Author: I'll remember this for the next z-streams. Thanks!

> * CENTOS 8 has recently ended its lifecycle. This release updates the CENTOS 8 references to CENTOS 9. (link:https://issues.redhat.com/browse/OCPBUGS-39160[*OCPBUGS-39160*])

Reviewer: The [OCPBUGS-39160](https://issues.redhat.com/browse/OCPBUGS-39160) cited here is a 4.14.z bug already shipped/mentioned in the [4.14 RNs](https://docs.openshift.com/container-platform/4.14/release_notes/ocp-4-14-release-notes.html#ocp-4-16-36-enhancements-_release-notes). It looks like [OCPBUGS-41786](https://issues.redhat.com/browse/OCPBUGS-41786) is the 4.13 version of this bug, and appears to be attached to the actual 4.13.z advisory.

Author: :heavy_check_mark:

[id="ocp-4-13-51-bug-fixes"]
==== Bug fixes

> * Previously, `openvswitch` could function by using older configurations after a cluster upgrade operations. These older configurations might have caused issues for networking components that interact with `openvswitch`. With this release, `openvswitch` is restarted after an upgrade operation so that `openvswitch`  is forced to receive the latest configurations. (link:https://issues.redhat.com/browse/OCPBUGS-41723[*OCPBUGS-41723*])

Reviewer: Typo + white space, but it also seems like it's mostly referred to across the repo generically as "Open vSwitch (OVS)":
```suggestion
* Previously, Open vSwitch (OVS) could function by using older configurations after cluster upgrade operations. These older configurations might have caused issues for networking components that interact with OVS. With this release, OVS is restarted after an upgrade operation so that OVS is forced to receive the latest configurations. (link:https://issues.redhat.com/browse/OCPBUGS-41723[*OCPBUGS-41723*])
```

> * Previously, the Cluster Resource Override Operator was incorrectly using `kube` version 1.21 clients on {product-title}. The Operator should have been using `kube` version 1.26 clients. The `kube` version 1.21 clients can no longer work because the `flowcontrol.apiserver.k8s.io/v1beta1` API was previously removed. With this release, the Cluster Resource Override Operator now uses `kube` version 1.21 clients so that this issue no longer persists. (link:https://issues.redhat.com/browse/OCPBUGS-39236[*OCPBUGS-39236*])

Reviewer: Does it need to explicitly reference it as "`kube`" like this, or can this just be generically "Kubernetes"? E.g.:
```suggestion
* Previously, the Cluster Resource Override Operator was incorrectly using Kubernetes version 1.21 clients on {product-title}. The Operator should have been using Kubernetes version 1.26 clients. Kubernetes 1.21 clients can no longer work because the `flowcontrol.apiserver.k8s.io/v1beta1` API was previously removed. With this release, the Cluster Resource Override Operator now uses Kubernetes 1.21 clients so that this issue no longer persists. (link:https://issues.redhat.com/browse/OCPBUGS-39236[*OCPBUGS-39236*])
```

Author: :heavy_check_mark:

Space precluded documenting all of the container images for this release in the advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.13.51 --pullspecs
----

[id="ocp-4-13-51-enhancements"]
==== Enhancements

* CentOS 8 has recently ended its lifecycle. This release updates the CentOS 8 references in the {product-title} installer to CentOS 9. (link:https://issues.redhat.com/browse/OCPBUGS-39160[*OCPBUGS-39160*])

[id="ocp-4-13-51-bug-fixes"]
==== Bug fixes

* Previously, Open vSwitch (OVS) could function by using older configurations after a cluster upgrade operations. These older configurations might have caused issues for networking components that interact with OVS. With this release, OVS is restarted after an upgrade operation so that OVS is forced to receive the latest configurations. (link:https://issues.redhat.com/browse/OCPBUGS-41723[*OCPBUGS-41723*])

> * Previously, the Cluster Resource Override Operator was incorrectly using Kubernetes version 1.21 clients on {product-title}. The Operator should have been using Kubernetes version 1.26 clients. The Kubernetes version 1.21 clients can no longer work because the `flowcontrol.apiserver.k8s.io/v1beta1` API was previously removed. With this release, the Cluster Resource Override Operator now uses Kubernetes version 1.21 clients so that this issue no longer persists. (link:https://issues.redhat.com/browse/OCPBUGS-39236[*OCPBUGS-39236*])

Author: Remove.

Author: ![Screenshot from 2024-09-24 10-02-08](https://github.com/user-attachments/assets/30518128-4f70-497c-8f5f-e6ecfb91b6c6)
